pkg/master/controller.go

```diff
@@ -240,10 +248,23 @@ func (c *Controller) CreateOrUpdateMasterServiceIfNeeded(serviceName string, ser
 	return err
 }
 
-// ReconcileEndpoints sets the endpoints for the given apiserver service (ro or rw).
-// ReconcileEndpoints expects that the endpoints objects it manages will all be
-// managed only by ReconcileEndpoints; therefore, to understand this, you need only
-// understand the requirements and the body of this function.
+// ReconcileEndpoints sets the endpoints for the given apiserver service (ro
+// or rw).  It is a wrapper that calls ReconcileEndpointsUsingMasterCount
+// unless the controller is configured to use master leases, in which case, it
+// calls ReconcileEndpointsUsingLeases.
+func (c *Controller) ReconcileEndpoints(serviceName string, ip net.IP, endpointPorts []api.EndpointPort, reconcilePorts bool) error {
+	if c.MasterLeases != nil {
```

Comment:
Consider making two implementations of the same interface instead of if/switches like this?